pyflakes/checker.py

```diff
@@ -1160,6 +1166,14 @@ def addBinding(self, node, value):
         # in scope
         if value.name not in self.scope or not isinstance(value, Annotation):
             self.scope[value.name] = value
+            scope = self.scope
+            # As per PEP 572, use scope in which Generator is defined
+            if (
+                isinstance(value, NamedExprAssignment) and
+                isinstance(self.scope, GeneratorScope)
+            ):
+                scope = self.scopeStack[-2]
```

Comment:
does this need special handling for `nonlocal` / `global` ?

Reply:
@asottile I don't think so? I think the caveats pointed out in the PEP about nonlocal/global regard which variables get assigned. But since pyflakes just does static analysis, I don't think it is relevant. I tested it with the following snippet:
```python
global_var = False


def global_keyword_test():
    global global_var
    if any(global_var := f[0] for f in [[True]]):
        print(global_var)


def global_no_keyword_test():
    if any(global_var := f[0] for f in [[True]]):
        print(global_var)


def nonlocal_keyword_test():
    local_var = False

    def helper():
        nonlocal local_var
        if any(local_var := f[0] for f in [[True]]):
            print(local_var)
    helper()
    print(local_var)


def nonlocal_no_keyword_test():
    local_var_2 = False

    def helper():
        if any(local_var_2 := f[0] for f in [[True]]):
            print(local_var_2)
    helper()
    print(local_var_2)


def original_issue():
    def helper():
        if any(local_var_v2 := f[0] for f in [[True]]):
            print(local_var_v2)
    helper()


print("global_keyword_test")
global_keyword_test()
print(global_var)
print("")
global_var = False

print("global_no_keyword_test")
global_no_keyword_test()
print(global_var)
print("")
global_var = False

print("nonlocal_keyword_test")
nonlocal_keyword_test()
print("")

print("nonlocal_no_keyword_test")
nonlocal_no_keyword_test()
print("")
```
And neither `master`, nor this branch, had any issues with any of the functions, except for `master`, which reported the F821 for `original_issue`

The output was as follows:
```
global_keyword_test
True
True

global_no_keyword_test
True
False

nonlocal_keyword_test
True
True

nonlocal_no_keyword_test
True
False
```